dojo/templatetags/get_endpoint_status.py

```diff
@@ -15,6 +15,16 @@ def get_mitigated_endpoints(finding):
     return [status.endpoint for status in status_list]
 
 
+@register.filter
+def vulnerable_endpoint_count(finding):
+    return finding.endpoint_status.all().filter(mitigated=False).count()
+
+
+@register.filter
+def mitigated_endpoint_count(finding):
+    return finding.endpoint_status.all().filter(mitigated=True).count()
+
+
```

Comment:
The tags do use `finding.endpoint_status.all()`. Are you saying it would be quicker converting a list and getting the length than one query to the db by using `.count()`?

Unfortunately checking the status of the the `endpoint_status` object requires both the endpoint and the finding, so I'm not sure how to prefetch with that one. Both `endpoints` and `endpoint_status` objects are prefetched, so maybe it works?

Reply:
the filter() will trigger a new query so you would have to filter in python. and also count in python (with len or so) because count would trigger a query even without filter(). This is exactly what happens if you use prefetching, but then for all rows at once and transparantly by django orm.

I just tried it in here, and the prefetch just works for me:

> prefetched_findings = prefetched_findings.annotate(active_endpoint_count=Count(endpoint_status__id', filter=Q(endpoint_status__mitigated=False)))

But then again I don't use endpoitn usually. I added some via the UI but no matter what I do, there are never any entries in the `dojo_finding_endpoint_status`  table, only in the `dojo_endpoint_status` table so the count is always zero.

Reply:
I wonder if it doesn’t quite work on the API then. Sounds like creation is just fine, but establishing the link between endpoints and findings isn’t quite working.